src/parser/core/modules/CooldownDowntime.js

```diff
@@ -0,0 +1,183 @@
+import React from 'react'
+import {getAction} from 'data/ACTIONS'
+import Module from 'parser/core/Module'
+import {Rule, Requirement} from 'parser/core/modules/Checklist'
+import {ActionLink} from 'components/ui/DbLink'
+
+export default class CooldownDowntime extends Module {
+	static handle = 'cooldowndowntime'
+	static dependencies = [
+		'combatants',
+		'cooldowns',
+		'suggestions',
+		'checklist',
+		'downtime',
+	]
+
+	//Downtime Actions IDs
+	_downTime = {}
+	//tracking the important™ CDs
+	_trackedcds = []
+	//Default alloted time before a spell is held too long.
+	_downtimeOkTime = 0
+	//Determines if we're being overriden by an extension class
+	_extensionOverride = false
+
+	constructor(...args) {
+		super(...args)
+
+		this.addHook('begincast', {by: 'player'}, this._onBegin)
+		this.addHook('cast', {by: 'player'}, this._onCast)
+		this.addHook('complete', this._onComplete)
+	}
+
+	/**
+	 *This just sets the internal ability list if you aren't going to use override functions and
+	 perform calculated data holding in your extended class
+	 * @param {List of Abilities to track} abilities
+	 */
+	setAbilityList(abilities) {
+		this._trackedcds = abilities
+	}
+
+	/**
+	 * This instructs the module how long a CD can be held for before it is
+	 * problematic to hold.
+	 * @param {Time in MS allowed to hold a CD for} duration
+	 */
+	setDownTimeDuration(duration) {
+		this._downtimeOkTime = duration
+	}
+
+	/**
+	 * Denotes if we're overriding this super from an extension
+	 * @param {boolean, whether or not we override this super from an extension} override
+	 */
+	setExtensionOverride(override) {
+		this._extensionOverride = override
+	}
+
+	_onBegin(event) {
```

Comment:
Both of these are just passing over to the check function - you can probably just pass `this._cooldownCheck` straight into the hooks up top.